### PR TITLE
feat: BNGP-4543 Credits Purchase journey feature flag

### PR DESCRIPTION
**Pull request by KiranVarma02**

### CODE REVIEW

File: packages/webapp/src/credits/constants.js
Comment (pwadmore-ea) [inline packages/webapp/src/credits/constants.js:1]: As agreed with @paulsimonandrews, this file needs to be split into distinct files for credits estimation and credits purchase located in the utils directory:
- packages/webapp/src/utils/credits-estimation-constants.js
- packages/webapp/src/utils/credits-purchase-constants.js

File: packages/webapp/src/routes/credits-purchase/credits-individual-or-organisation.js
Comment (pwadmore-ea) [inline packages/webapp/src/routes/credits-purchase/credits-individual-or-organisation.js:1]: As the copydeck for this functionality has not been issued I think it is sensible to follow the naming conventions used by the copydecks for the landowner and developer journeys:

- credits-purchase/applying-individual-organisation.js

File: packages/webapp/src/views/credits-purchase/credits-individual-or-organisation.html
Comment (pwadmore-ea) [inline packages/webapp/src/views/credits-purchase/credits-individual-or-organisation.html:1]: As the copydeck for this functionality has not been issued I think it is sensible to follow the naming conventions used by the copydecks for the landowner and developer journeys:

- credits-purchase/applying-individual-organisation.html

**Review by pwadmore-ea — CHANGES_REQUESTED** [review]
Small changes requested concerning file names and locations.

File: packages/webapp/src/routes/credits-purchase/credits-individual-or-organisation.js
Comment (KiranVarma02) [reply packages/webapp/src/routes/credits-purchase/credits-individual-or-organisation.js:1]: this is not part of this ticket. Do I make this change here or remove it completely?

I have only added it to test as we were discussing other day.

Comment (pwadmore-ea) [reply packages/webapp/src/routes/credits-purchase/credits-individual-or-organisation.js:1]: As discussed, please make the change here and keep this minimal route so QA have something to test the feature flag with. https://eaflood.atlassian.net/browse/BNGP-2880 can replace the dummy implementation when ready.

File: packages/webapp/src/views/credits-purchase/credits-individual-or-organisation.html
Comment (pwadmore-ea) [reply packages/webapp/src/views/credits-purchase/credits-individual-or-organisation.html:1]: As discussed, please make the change here and keep this minimal view so QA have something to test the feature flag with. https://eaflood.atlassian.net/browse/BNGP-2880 can replace the dummy implementation when ready.

File: packages/webapp/src/utils/credits-estimation-constants.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  Object.entries(creditEstimateRoutes).map(([k, v]) => [k, v.substring(1)])
+)
+
+const redisKeys = {
+  ESTIMATOR_CREDITS_CALCULATION: 'estimator-credits-calculation'
+}
+
+export default {
+  CREDITS_ESTIMATION_PATH,
+  creditEstimateRoutes,
```
Comment (pwadmore-ea) [inline packages/webapp/src/utils/credits-estimation-constants.js:18]: Please replace this with routes following the split into distinct files for credits estimation and credits purchase.

File: packages/webapp/src/utils/credits-purchase-constants.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+const creditPurchaseRoutes = {
+  ESTIMATOR_CREDITS_INDIVIDUAL_ORG: '/credits-purchase/credits-individual-or-organisation'
+}
+
+const views = Object.fromEntries(
+  Object.entries(creditPurchaseRoutes).map(([k, v]) => [k, v.substring(1)])
+)
+
+export default {
+  creditPurchaseRoutes,
```
Comment (pwadmore-ea) [inline packages/webapp/src/utils/credits-purchase-constants.js:10]: Please replace this with routes following the split into distinct files for credits estimation and credits purchase.

File: packages/webapp/src/utils/constants.js [hunk 3]
```diff
@@ -1,7 +1,8 @@
 import developerConstants from './developer-constants.js'
 import { NODE_ENV, AZURE_FUNCTION_APP_URL } from './config.js'
 import lojConstants from './loj-constants.js'
-import creditsConstants from '../credits/constants.js'
+import creditsEstimationConstants from './credits-estimation-constants.js'
+import creditPurchaseRoutes from './credits-purchase-constants.js'
```
Comment (pwadmore-ea) [inline packages/webapp/src/utils/constants.js:5]: Please correct linting error.

**Review by pwadmore-ea — CHANGES_REQUESTED** [review]
Please see further review comments

File: packages/webapp/src/credits/constants.js
Comment (KiranVarma02) [reply packages/webapp/src/credits/constants.js:1]: done

File: packages/webapp/src/utils/credits-estimation-constants.js (hunk 1 above)
Comment (KiranVarma02) [reply packages/webapp/src/utils/credits-estimation-constants.js:18]: done

File: packages/webapp/src/routes/credits-purchase/credits-individual-or-organisation.js
Comment (KiranVarma02) [reply packages/webapp/src/routes/credits-purchase/credits-individual-or-organisation.js:1]: done

File: packages/webapp/src/views/credits-purchase/credits-individual-or-organisation.html
Comment (KiranVarma02) [reply packages/webapp/src/views/credits-purchase/credits-individual-or-organisation.html:1]: done

**Review by pwadmore-ea — APPROVED** [review]

**Comment by shaundefra** [comment]
tested by Tajinder

**Pull request merged**